docs/adhara/ethereum_proof_verification.md

```diff
@@ -0,0 +1,167 @@
+
+Every QBFT block header is signed by (a subset of) validators of the network, and thus the signatories need to be compared to a known list of validators, as well as ensure that enough signatures are present to exceed a certain threshold, for a given network.
+
+In addition, the Merkle tree root would need to be recalculated, using the event in question, and then verify that this root matches the transaction receipt root in the previously verified block header. Recalculating the Merkle tree root only requires a subset (only sibling and parent nodes of the transaction receipt log containing the event in question) of the tree, very similar to how a Merkle inclusion proof would be verified.
+
+## Introduction
+
+This section describes how Ethereum events can be turned into EEA-compliant crosschain function calls, that is cryptographically secured by Ethereum event attestation proofs, using block headers.
+
+An Ethereum block header consists of the following fields:
+
+1. `parentHash`: The Keccak 256-bit hash of the parent block's header.
+2. `sha3Uncles`: The Keccak 256-bit hash of the ommers list portion of this block.
+3. `miner`: Miner who mined the block.
+4. `stateRoot`: The Keccak 256-bit hash of the root node of the state trie.
+5. `transactionsRoot`: The Keccak 256-bit hash of the root node of the transaction trie.
+6. `receiptsRoot`: The Keccak 256-bit hash of the root node of the receipts trie.
+7. `logsBloom`: The Bloom filter composed out of information contained in each log from the receipt of each transaction.
+8. `difficulty`: A scalar value corresponding to the effort required to mine the block.
+9. `number`: A scalar value equal to the number of ancestor blocks.
+10. `gasLimit`: A scalar value equal to the current limit of gas expenditure per block.
+11. `gasUsed`: A scalar value equal to the total gas used in transactions in this block.
+12. `timestamp`: A scalar value equal to the time at which the block was mined.
+13. `extraData`: An arbitrary byte array containing data relevant to this block.
+14. `mixHash`: A 256-bit hash. When combined with the nonce proves that a sufficient amount of work has been carried out on this block.
+15. `nonce`: A 256-bit hash. When combined with the mixHash proves that a sufficient amount of work has been carried out on this block.
+
+The verification of an EEA-compliant Ethereum block header proof consists of:
+
+1. verifying that the block header and block header preimage match up and hash to the block hash that was signed over.
+2. verifying that the transaction receipts root extracted from the verified block header match up with the root provided in the proof.
+3. verifying a Merkle inclusion proof for the given logs in the receipts tree.
+4. verifying the provided validator signatures over the block header containing the receipts tree root.
+5. verifying that at least half the onboarded validators' signatures are present.
```

Comment:
```suggestion
5. verifying that a super-majority (greater than or equal to 2/3) of the onboarded validators' signatures are present.
```
https://docs.goquorum.consensys.io/configure-and-manage/configure/consensus-protocols/qbft

Reply:
Do you want to change the code as well?

Reply:
Will correct.